libparsec/crates/client/src/connection_monitor.rs

```diff
@@ -84,20 +84,47 @@ impl ConnectionMonitor {
     pub async fn start(cmds: Arc<AuthenticatedCmds>, event_bus: EventBus) -> Self {
         let worker = spawn(async move {
             let mut state = ConnectionState::Offline;
-            let mut stream = cmds.start_sse::<Req>();
+            let mut last_event_id: Option<String> = None;
+            let mut backoff = BackoffLimited::new();
 
             loop {
-                match stream.next().await {
-                    Ok(SSEResponseOrMissedEvents::MissedEvents) => {
-                        event_bus.send(&EventMissedServerEvents);
-                    }
-                    Ok(SSEResponseOrMissedEvents::Response(rep)) => {
-                        handle_sse_response(&mut state, &event_bus, rep);
-                    }
+                backoff.wait().await;
+
+                let mut stream = match cmds.start_sse::<Req>(last_event_id.as_ref()).await {
+                    Ok(stream) => stream,
                     Err(err) => {
-                        if handle_sse_error(&mut state, &event_bus, err).is_break() {
+                        if handle_sse_error(&mut state, &event_bus, err.into()).is_break() {
                             return;
                         }
+                        continue;
+                    }
+                };
+
+                backoff.reset();
+
+                while let Some(res) = stream.next().await {
+                    match res {
+                        Ok(event) => {
```

Comment:
when connection is back online, `backoff`'s attempt must be reset

Reply:
backoff attempt is indeed reset on line `103`